src/coord/src/coord.rs

```diff
@@ -540,18 +561,25 @@ where
                 }
             }
 
+            let needed = self.need_advance;
             let mut next_ts = self.get_ts();
+            self.need_advance = false;
             if next_ts <= self.read_lower_bound {
                 next_ts = self.read_lower_bound + 1;
             }
-            if next_ts > self.closed_up_to {
-                broadcast(
-                    &mut self.broadcast_tx,
-                    SequencedCommand::AdvanceAllLocalInputs {
-                        advance_to: next_ts,
-                    },
-                );
-                self.closed_up_to = next_ts;
+            // TODO(justin): this is pretty hacky, and works more-or-less because this frequency
+            // lines up with that used in the logging views.
+            const FREQUENCY: u64 = 1_000;
+            if needed || next_ts / FREQUENCY > self.closed_up_to / FREQUENCY {
```

Comment:
So the only thing that worries me here is that `--logging-frequency` is a thing we can change at the command line. By default it is 1s, but if someone dials it up or down I don't know what will happen. We can probably punt that off to the future, but it's the sort of thing that will accrete with time. I don't have anything better (maybe steal the same argument from `--logging-frequency` but wanted to surface shared concern.

Reply:
Plumbed that value through to here as well. Kinda clunky but I think better!